### ISSUE TITLE
Running mean and variance with `tf.keras.layers.Normalization`

### ISSUE THREAD

**Issue by markub3327** [keras-team/keras#17287]
**System information**.

TensorFlow version (you are using): 2.10.1
Are you willing to contribute it (Yes/No) : Yes

**Describe the feature and the current behavior/state**.

Can `tf.keras.layers.Normalization` be used in running mean and variance manner? 
I need adapt this layer as the first step in the custom training function during training.

Is effectively runs `adapt(data, batch_size=256, steps=1)` with 1 step on the whole batch at every `def train_step(self, data):`? 

Or better way is firstly runs `adapt(data, batch_size=256, steps=1)` before training and during training (in `def train_step(self, sample):`) runs only `update_state(data)` and then `finalize_state()`, what can have less Python overhead?

**Will this change the current api? How?**
Maybe no.

**Who will benefit from this feature?**
Reinforcement Learning

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**

- Do you want to contribute a PR? (yes/no): no
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):

**Comment by markub3327** [comment]
Sorry, I missed.

The `adapt` method cannot be used inside XXX:
```python
"Detected a call to `PreprocessingLayer.{method_name}` inside a "
            "`tf.function`. `PreprocessingLayer.{method_name} is a high-level "
            "endpoint that manages its own `tf.function`. Please move the call "
            "to `PreprocessingLayer.{method_name}` outside of all enclosing "
            "`tf.function`s. Note that you can call a `PreprocessingLayer` "
            "directly on `Tensor`s inside a `tf.function` like: `layer(x)`, "
            "or update its state like: `layer.update_state(x)`."
```

In this message is nothing about `finalize_state()`. All updates done inside `def train_step(self, sample):` using `layer.update_state(x)` are not applied to `self.mean` and `self.variance`.

---------------------------

Please inform the users in documentation about `finalize_state()` method, which must be used for example after `self.model.load_weights(model_path)`, `layer.update_state(x)`, and any update of normalization layer variables using `variable.assign(value)`.

Thanks.

**Comment by sushreebarsa** [comment]
@markub3327 Thank you for the update. Could you please confirm if this issue is resolved for you?
Thank you!

**Comment by markub3327** [comment]
Still no .... Please inform the users in documentation about `finalize_state()` method, which must be used for example after `self.model.load_weights(model_path)`, `layer.update_state(x)`, and any update of normalization layer variables using `variable.assign(value)`.

Sorry, this is not a feature request but the documentation issue. Can you change the type of this issue?

Thanks.
